src/java.desktop/share/classes/javax/swing/JPasswordField.java

```diff
@@ -60,6 +60,14 @@
  * If an application needs the input methods support, please use the
  * inherited method, <code>enableInputMethods(true)</code>.
  * <p>
+ * <strong>Warning:</strong> The text entered in {@code JPasswordField} displays
+ * something that was typed and does not show the original characters. This
+ * doesn't prevent the password from appearing in the system memory. For
+ * handling confidential information such as the password text, refer to the
+ * relevant section at
+ * <a href="https://www.oracle.com/java/technologies/javase/seccodeguide.html">
```

Comment:
We might want to use a link with an anchor to the specific section, e.g.:
https://www.oracle.com/java/technologies/javase/seccodeguide.html#2-2

Reply:
If we want to allow that the representation of what was typed is anything the UI decides then let's start from scratch,

The {@code JPasswordField} will not show the original characters that were typed, instead displaying alternative text or graphics.
However this doesn't prevent... [and then the rest is as originally proposed]